add iptables physdev option

mwellmeyer opened the pull request:
##### SUMMARY
Add the opportunity to filter within bridge interfaces.
The physdev function is required if you want to add iptables rules, which should filter the traffic for an interface that is enslaved to a bridge.

##### ISSUE TYPE
 - Feature Pull Request

##### COMPONENT NAME
iptables module - lib/ansible/modules/system/iptables.py

##### ANSIBLE VERSION
<!--- Paste verbatim output from "ansible --version" between quotes below -->
```
ansible 2.5.0
```

mwellmeyer committed:
add iptables physdev option

mwellmeyer committed:
add iptables physdev option

mwellmeyer committed:
add iptables physdev option

mwellmeyer committed:
add iptables physdev option

sebastiendarocha commented:
Hello @mwellmeyer 

Could you rebase your PR ?

http://docs.ansible.com/ansible/latest/dev_guide/developing_rebasing.html

Commit:
fix missing blank lines

Commit:
fix missing blank lines

mwellmeyer commented:
rebased

bcoca committed:
better error on empty playbook

fixes #45480

dagwieers committed:
Ensure _raw_params retain exact spaces

This fixes a few issues related to multi-line YAML strings that eat
whitespace and add whitespaces after newlines (that weren't there).

smnmtzgr committed:
added new module aci_access_port_block_to_access_port

dagwieers committed:
Add integration tests

By checking the cmd result with the original we cause these tests to
fail on older releases without this PR.

dagwieers committed:
Add unit tests

dagwieers committed:
Add less judgmental hint to use script + template

s-hertel committed:
Fix _uri option documentation for cache plugins that expect strings

smnmtzgr committed:
changed documentation strings

mwellmeyer committed:
Merge branch 'devel' into add_physdev

Commit:
wrong ansible version

Commit:
wrong ansible version

joachimmathes committed:
Generalize config retrieval in ios_l3_interface module (#45270)

This makes it work with Cisco 3750E.

gustavomcarmo committed:
Fix iterator to list conversion in ldap_entry module

PR #45778 fixes #45417

akira6592 committed:
The sshkey option is available in version 2.7 and above (#46103)

bankabanka committed:
Update sl_vm.py (#43966)

* Update sl_vm.py

Updated the list of datacenters to include new datacenters.

tomww committed:
Update synchronize.py, mention variable ansible_rsync_path needed for different local rsync binary (#41870)

* Update synchronize.py

If you want a different rsync binary on the local side, you need to set task variable ansible_rsync_path.
See examples.

Variable ansible_rsync_path looks to not be documented anywhere. If documented, needs to be said that is does not belong to synchronise options, instead belongs to tasks. (Sorry, I have no better wording)

Hirshr committed:
spotinst_aws_elastigroup elastic_ips example (#41190)

* Update spotinst_aws_elastigroup.py

dmurarasu committed:
fixed nxos_aaa_server_host issue with type 7 encrypt key (#46015)

* fixed nxos_aaa_server_host issue with type 7 encrypt and added test cases

* added idempotence tests

newtonne committed:
homebrew_cask: Pass install_options during uninstall (#44938)

Until https://github.com/Homebrew/homebrew-cask/issues/40866 is fixed,
install_options should be passed when uninstalling casks to ensure that
all artefacts are removed.

acozine committed:
restores legacy testing file (#46095)

* restores legacy testing file

* corrects links to legacy testing page

* oops, copy-pasta fix

Akasurde committed:
[docs] Fix typos in aws_s3 and s3_bucket module docs (#46106)

Eucalypus to Eucalyptus

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

resmo committed:
cs_instance: fix host migration without volume (#46115)

sivel committed:
pylint plugin to catch due/past-due deprecated calls (#44143)

* Start of work on pylint plugin to catch due/past-due deprecated calls

* Improve deprecated pylint plugin

* Catch call to AnsibleModule.deprecate also

* Skip splatted kwargs, we can't infer that info

* Add error for invalid version in deprecation

* Skip version if it's a reference to a var

* Disable ansible-deprecated-no-version for displaying deprecated module info

* fix comments

* is None

* Force specifying a version, this can be disabled on a per case basis

* Disable ansible-deprecated-version by default

* Remove to look for 2.8 deprecated

* Revert "Remove to look for 2.8 deprecated"

This reverts commit 4e84034fd104879f429f0262ff0b2317e3d08deb.

* Add script and template used for creating issues for deprecated issues

* Fix underscore var

abarbare committed:
feat: add scaleway security group module (#45686)

fix: pep8 coding style

fix: error order requests

remyleone committed:
Add a guide for Scaleway (#44810)

gundalow committed:
ROADMAP_2_8 stub (#46120)

* ROADMAP_2_8 stub

* Update ROADMAP_2_8.rst

Commit:
fix for managed disk facts (#46021)

gundalow committed:
botmeta support: core (#45917)

* botmeta enforce `support: core`

bcoca committed:
allow jinja2 unique filter compat (#45637)

* allow jinja2 unique filter compat
* detect if unique is provided, fallback with warning
* handle j2 specific params
* now all filters using unique must pass environment
* added env to tests

also normalized on how we normally import and use exceptoins

josemaia committed:
[azure] Fix name for the symbol * in docs (#46119)

Correct name is "asterisk", not "asterix", which is a popular franco-belgian comic character!

+label: docsite_pr

jborean93 committed:
skip installing linux-image-extra in CI as it shouldn't be needed (#46101)

pilou- committed:
Speed up VariableManager by preserving Templar state. (#45572)

Maintain one Templar for the lifetime of VariableManager, calling
set_available_variables() prior to each use, enabling _get_filter()'s
cache to function correctly.

It does not seem possible for concurrent calls into one (non-copied)
VariableManager instance, and so it need not be reentrant. If that
became a requirement, serializing its or Templar's entry points would be
fine, as it's so CPU-heavy other threads will only fight with it for the
GIL anyway.

Reduces _get_filters() runtime 91%, get_vars() runtime 19%, function
call count 16%, overall runtime 10%.

Tested aginst dummy load comprised of the 12 disabled steps of
debops.auth with an inventory of 80 hosts, which stresses variable
processing and task setup. Before:

         7447296 function calls (7253994 primitive calls) in 32.611 seconds

   Ordered by: cumulative time

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
        1    0.000    0.000   32.762   32.762 ansible-playbook:3(<module>)
        1    0.007    0.007   31.733   31.733 ansible-playbook:21(<module>)
    ...
 1371/971    0.671    0.000   21.332    0.022 manager.py:154(get_vars)
    ...
     3044    0.315    0.000    5.166    0.002 __init__.py:295(_get_filters)

After:

         6252978 function calls (6059638 primitive calls) in 29.055 seconds

   Ordered by: cumulative time

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
        1    0.000    0.000   29.218   29.218 ansible-playbook:3(<module>)
        1    0.007    0.007   28.159   28.159 ansible-playbook:21(<module>)
    ...
 1371/971    0.675    0.000   17.211    0.018 manager.py:154(get_vars)
    ...
     3044    0.028    0.000    0.441    0.000 __init__.py:295(_get_filters)

Commit:
EC2_ASG: Enable support for launch_templates (#45647)

* Enable support for launch_templates in ec2_asg

* Fix asg create with LT and no version number

* Update mutually exclusive list

* Better function names

ryansb committed:
[docs] Add changelog for ec2_asg launch template feature

jborean93 committed:
win_setup: don't fail when running as non-admin and EAP is set to Stop (#46149)

Akasurde committed:
VMware: Add support for service type (#40651)

* Add support for - Replication, Provisioning
* Update documentation
* Update example

Fixes: #40529

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

Akasurde committed:
VMware: trigger customization for customization_spec (#45415)

This fix allows to trigger VM customization if user specifies
customization spec.

Fixes: #44159

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

felixfontein committed:
docker-py: bump minimal version to 1.8.0 (#46134)

* Bumping docker-py minimal version to 1.8.0.
* Improve docs formatting.

resmo committed:
cs_instance: doc: fix typo in examples (#46035)

resmo committed:
cs_staticnat: fix sanity (#46037)

resmo committed:
cs_ip_address: use query_api, fixes error handling (#46034)

resmo committed:
cs_resourcelimit: use query_api for error handling (#46036)

Commit:
fix for banner idempotent issue (#46001)

* fix for banner issue

* review comments

tstoner committed:
Added feature facts to nxos_facts (#45934)

* Added feature facts to nxos_facts

* Fixed ansibot indentation

* Resolved ansibot whitespace missing after ','

* Per PR suggestion, created method in base class to gather switch config
and store in global variable to prevent multiple calls to switch.

* Addressed ansibot blank line & whitespace after ,

aruhier committed:
Bugfixes in Netbox inventory (#45731)

* Fix headers and params in netbox queries

Fix token authentication
Remove unused variables regarding api params and headers

* Fix python 3 incompatibility in netbox inventory

* Cleaning BOTMETA entry

samccann committed:
made sros_config docs match code for match options (#46135)

* made docs match code for match options

* fixed shippable error

aaronk1 committed:
Update win_robocopy.py (#38633)

* Update win_robocopy.py

<!--- Show example of UNC path copy of a single file, noting trailing backslash required -->

+label: docsite_pr

* addresses jborean comment, removes double quotes

* Remove trailing slash part

smnmtzgr committed:
be compliant to pep8 and pylint. Added integration tests.

fabianvf committed:
add maintainers for k8s lookup and inventory plugins (#46180)

smnmtzgr committed:
added parameter leaf_port_blk_description in module and integration tests

pilou- committed:
openshift inventory plugin: fix exception when auth fails (#45826)

* openshift inventory: fix exception when auth fails

Fix 'ForbiddenError' object has no attribute 'message':

    [WARNING]:  * Failed to parse test.yml with openshift plugin: 'ForbiddenError' object has no attribute 'message'
     File "ansible/lib/ansible/inventory/manager.py", line 270, in parse_source
       plugin.parse(self._inventory, self._loader, source, cache=cache)
     File "ansible/lib/ansible/plugins/inventory/openshift.py", line 122, in parse
       self.setup(config_data, cache, cache_key)
     File "ansible/lib/ansible/module_utils/k8s/inventory.py", line 58, in setup
       self.fetch_objects(connections)
     File "ansible/lib/ansible/module_utils/k8s/inventory.py", line 250, in fetch_objects
       super(OpenShiftInventoryHelper, self).fetch_objects(connections)
     File "ansible/lib/ansible/module_utils/k8s/inventory.py", line 81, in fetch_objects
       namespaces = self.get_available_namespaces(client)
     File "ansible/lib/ansible/module_utils/k8s/inventory.py", line 95, in get_available_namespaces
       raise K8sInventoryException('Error fetching Namespace list: {0}'.format(exc.message))

Don't try to get 'message' attribute from:
- K8sInventoryException instances
- Exception instances
- KubernetesException instances (because KubernetesException can be
  Exception)

* move k8s/OpenShift inventory plugin dedicated code

inventory plugin specific code should not be located in
lib/ansible/module_utils directory. Then ansible.utils methods can be
reused (for example Display).

* Remove unused class variables 'helper'

unused since 4d77878654e867c23d5f6c61422bdae7120393bc.

yungezz committed:
add azure_rm_cdnprofile and azure_rm_cdnprofile_facts (#45097)

* add cdn profile with facts

* fix merge issue

* fix version

* fix lint

* move cdn client out of azure_rm_common

* fix lint

* fix helper

* fix code bug

* fix bug

* fix import

* fix typo

* fix test

ksubileau committed:
win_nssm: tests and several bug fixes (#44755)

* win_nssm: add failing tests for issue #44079

* win_nssm: use Run-Command instead of Invoke-Expression to prevent interpretation issue

Fix #44079

* win_nssm: add more failing tests

These tests highlight several issues with this module:
 * Service not started when state=started
 * Errors with app_parameters (see #25265)
 * Exception when passing several dependencies separated by comma as specified in doc

* win_nssm: fix service not started when state=started

Nssm status returns a multiline output that doesn't match any of the strict patterns in the switch statement.

* win_nssm: fix incorrect separator in doc for service dependencies

The dependencies parameter works with space as separator, but not with comma as shown in the documentation

* win_nssm: fix error with app_parameters parameter

Fix #25265

* win_nssm: add idempotence tests

* win_nssm: fix several idempotence issues and misbehaviors

Add missing space between arguments when app_parameters contains several keys.
Use Argv-ToString and Escape-Argument to improve arguments handling (parameters with quotes, backslashes or spaces).

* win_nssm: test parameters with spaces, quotes or backslashes

* win_nssm: restore comma as separator for service dependencies

Revert commit ddd4b4b

* win_nssm: restore support of string as dict form for app_parameters and remove support of literal YAML dict

* win_nssm: wrong variable in tests

dpassante committed:
cs_account: Implement role parameter support (#46166)

maxamillion committed:
Handle dnf immutable mutable datatypes (#46176)

In DNF < 3.0 are lists, and modifying them works
In DNF >= 3.0 < 3.6 are lists, but modifying them doesn't work
In DNF >= 3.6 have been turned into tuples, to communicate that
modifying them doesn't work

Further explanation of this is available via Adam Williamson from
the Fedora QA Team.

    https://www.happyassassin.net/2018/06/27/adams-debugging-adventures-the-immutable-mutable-object/

Signed-off-by: Adam Miller <admiller@redhat.com>

devyanikota committed:
Adding maintainer for GlusterFS modules. (#46160)

Signed-off-by: Devyani Kota <dkota@redhat.com>

Akasurde committed:
Misc typo fix (#46199)

jinaj2 to jinja2

Signed-off-by: Abhijeet Kasurde <akasurde@redhat.com>

sivel committed:
Fail on attrs that don't work for includes (#46177)

* Fail on attrs that don't work for includes

* Prefix with VALID_ to make it more clear

* Sort keywords

* Remove unnecessary list

* Adjust dynamic/static checking

dagwieers committed:
Clean up common module docs (#45888)

Some of the older modules have evolved documentation that does not
follow a consistent style. This is a clean-up in form and style.

dericcrago committed:
moved 'strftime' filter into 'Other Useful Filters' section (#45801)

samdoran committed:
Correct loop syntax in example (#46178)

ssahani committed:
nmcli: set hairpin to str to bool (#45908)

Fix: nmcli module always sets hairpin mode to 1 for bridge slaves.

Closes #42569

Signed-off-by: Susant Sahani<susant@redhat.com>

dagwieers committed:
Docs: Keep the module index clean (#46155)

A lot of modules have a short_description with a trailing dot even
though we don't want trailing dots in the index. This change removes
it when creating the document index.

felixfontein committed:
[aws] route53 module: fix idempotency for CAA records  (#46049)

* Fixing record order for CAA records to properly handle idempotency.

* Add integration tests that reproduce CAA failure

smnmtzgr committed:
added period in description. Changed integration test to work correctly.

mattclay committed:
Move module tests using TQM to integration tests.

abadger committed:
Update the minimum python versions to install (#46230)

* Update the minimum python versions to install

This will prevent modern pip from installing ansible-2.7.x in
python-2.6.  Users will get the Ansible-2.6.x maintenance releases
instead.

shinmog committed:
Adding connector for network/panos modules (#46142)

pilou- committed:
k8s/oc plugins: fix 'ResourceField' is not JSON serializable (#46145)

Fixes #44408

jborean93 committed:
Revert "powershell - fix for ANSIBLE_KEEP_REMOTE_FILES on older Pythons (#45942)" (#46238)

This reverts commit ce515a626c5d31fec55ab7b0304575d2bf7b9eb0.

s-hertel committed:
Fix ec2_group for EC2-Classic accounts (#46242)

* Fix ec2_group for EC2-Classic accounts

* changelog

felixfontein committed:
docker_container: allow to configure comparison for existing containers (#44789)

* Added comparison configuration.

* Improving user feedback on specifying a wrong option.

* Avoid bare except.

* Added basic integration tests.

* Adding wildcard support.

* Warn if ignore_image=yes is overridden.

* Added changelog fragment.

felixfontein committed:
No longer necessary since #33792 and #45628. (#45603)

machacekondra committed:
ovirt_host: Fix fail condititon of upgrade (#46104)

Signed-off-by: Ondra Machacek <omachace@redhat.com>

kbreit committed:
Meraki scenario guide - Changed section markup types (#46158)

* Changed section markup types, small fixes for style
- Add dashes per style guide
- Add TOC in case this grows

sdodsley committed:
Add label pure_storage for all related files. Add team_purestorage macro

direvus committed:
Clean up Vault docs in User Guide. (#46188)

* Fix spelling of 'separate' throughout.

* Various cleanups in the User Guide for Vault.

- Fix spelling of 'algorithm'
- Fix indentation of nested list in payload format
- Fix mysterious refernce to 'b_pkey1'.
- Fix reference to newline as '\n': the backslash is lost when rendered
to the docs website. Specify the hex value for newline instead of the
backslash escape.

* Fix formatting

* Update vault.rst

dbckz committed:
fix wording for silencing a sensu client (#46250)

vladimir-mencl-eresearch committed:
module/systemd: fix logic determining if a service needs to be enabled (#46245)

* modules/systemd: fix logic: allow scope to default to 'system'

Fix logic introduced in 7ea909418e5: if 'scope' param is not specified,
it defaults to system, but the value of module.params['scope'] is None,
not 'system' - so allow for that.

* modules/systemd: fix logic: disabled means disabled

Fix logic determining whether a service with both systemd and initd files is enabled or disabled.

In situations where systemd thinks service is disabled, but rc.d symlinks mark it as enabled,
this module wrongly assumes the service is enabled.

Fix this logic: disabled means disabled

Only when the output from does NOT include disabled, consider the status of rc.d symlinks.

This essentially replicates the fixes done to the systemd handling in the "service" module in 3c89a21e0c0

Fixes #22303

Fixes #44409

atrykush committed:
Corrected typo

Corrected typo
ansible_serach_path --> ansible_search_path
+label: docsite_pr

cscheib committed:
Add logstash_plugin example for lower heap size

In a scenario where you've set your logstash heap size high, you may need to set your heap lower in order to execute logstash-plugin while logstash is still running.

Here's an example of someone running into insufficient memory while executing logstash-plugin: https://discuss.elastic.co/t/logstash-plugin-and-jvm-options/99646

This scenario is probably fairly common, so it may be worth documenting as an example.

+label: docsite_pr

samdoran committed:
Use unicode instead of bytes (#46234)

The stdout and stderr values returned from self._low_level_execute() are text, not bytes. This results in an error in Python 3 since str and bytes cannot be concatenated.

Changing to unicode type allows this to work without error on Python 2 and Python 3.

acozine committed:
yes, we do run validate_modules in Shippable (#46280)

egroeper committed:
Update playbooks_delegation.rst (#46254)

Fix example variable lookup.

+label: docsite_pr

samdoran committed:
Use bytes rather than native string for result (#46281)

This prevents a stack trace in Python 3 when the result is an empty file since
the file is open in binary mode and a native string in Python 3 is str,
not bytes.

danieljakots committed:
Add support for OpenBSD (#46147)

bcoca committed:
avoid tests when ncclient is not installed

mkrupcale committed:
seboolean: Fix issue with changing persistent SELinux boolean values (#22779)

Previously, when the active SELinux value was the same as the desired value, regardless of the value of `persistent`, the module would simply exit successfully, assuming no need for update. This made the assumption that the active and persistent values should be the same, but that is not always the case. This modification treats both the active and persistent values separately when checking for the need to update. Note that the persistent update mode, however, will still update the active as well as the persistent value. If this is not desired, it is possible to separately toggle the active value alone.

Other changes:
* Make the check mode actually perform checks for changes
* Organizes semanage commands into set of logical steps

danieljaouen committed:
homebrew_cask: Fixes #38772

stromvirvel committed:
docker_container: ambiguous parameter "stop_timeout"  (#43874)

* docker_container: Honour stop_timeout when creating docker containers (#43814)

* Adjusting description to what actually happens.

See docker-py changelog for 2.7.0: 'APIClient.stop will
no longer override the stop_timeout value present in the
container’s configuration.'

* Add a test whether stop_timeout can be configured for the container.

* Added changelog.

* Integrate with comparisons (by default, ignore stop_timeout value for restarts; will be configurable with PR ansible/ansible#44789).

* Fix config change code and tests (#2)

* Improving wildcard test.
* Using correct config.

remyleone committed:
Add new member to botmeta

abarbare committed:
feat: add scaleway security_group_rule management (#45694)

* feat: add scaleway security_group_rule management

jmeridth committed:
Fix typo in ec2_vpnc_nat_gateway module (#46304)

- Added space between `format` and `in` on return item `create_time`

willthames committed:
Add the k8s label to the k8s files

k8s label for inventory plugin, module_utils
and modules

daBONDi committed:
Fix callout json plugin show global stats (#43123)

* fix callback Plugin json to support global stat set by set_stats module

* refactor to not break compatiblity, add trailing comma on output dict

* Remove sorted, as it not needed

* refactor, to sivels better code

* clean some code messup

* add changelog fragment

* added missing new line

* fix pep8 stuff

Asgavar committed:
Mention YAML as a AWS CloudFormation template language (#46345)

Qalthos committed:
Add configuration to override location of ansible-connection (#46128)

* Add configuration to override location of ansible-connection

* Check var or argv[0] before $PATH

queue-tip committed:
Use specified loopvar in include_role example loop

pilou- committed:
WIP: Check that union Jinja filter can be chained (#46298)

* Check that union Jinja filter can be chained

* set filters: fix unexpected templating type error

this error occurs with Jinja 2.10 since 32ec69d827393f6b1685bc3b089ea6c341b943e1,
for example when union filters are chained:

$ ansible all -i localhost, -mdebug -a"msg={{ []|union([])|union([]) }}"
localhost | FAILED! => {
    "msg": "Unexpected templating type error occurred on ({{ []|union([])|union([]) }}):
            unsupported operand type(s) for +: 'set' and 'list'"
}

Commit:
clean up doc - removed required: false (#46357)

jborean93 committed:
remove comments from multiplle yaml string that breaks tests for now

willthames committed:
Use recursive_diff for kubernetes (#45645)

Move recursive_diff from cloudfront_distribution to
common.dict_transformations and reuse it in k8s modules

heuels committed:
New module: routeros_facts – collect facts from RouterOS devices (#46114)

* feat(routeros): implement routeros_facts module

* fix(routeros): review by felixfontein

* fix(routeros): review by ganeshrn

* fix(routeros): review by felixfontein

mrmagooey committed:
vcenter_folder - fix error events being generated, incorrect parent folders (#45951)

* fix error events being generated when vmware folders exist
* fix incorrect parent folders being identified (wrong type)

cadl committed:
Ignore empty result of rabbitmqctl list_user_permissions (#35598)

Fix #34863

equelin committed:
Fix typo in the author's github username.

Ansibot is looking at the author's github username to query reviews for
modules in the same namespace.

peterjenkins1 committed:
Correct invalid example in jenkins_script docs (#46193)

acozine committed:
Updates release & maintenance version docs for 2.7

MatrixCrawler committed:
Add ssl support to consul_kv lookup (#42456)

* implemented lookup_consul_kv
* added missing version_added for consul_url ini section
* added default value for ANSIBLE_CONSUL_CLIENT_CERT and added some more documentation
* removed trailing whitespaces
* fixed indention
* Fixes in Documentation
* removed trailing whitespace
* removed trailing whitespace

rohansadale committed:
Add examples for bind and unmount to mount module docs

PR #46375

mattclay committed:
Rebalance shippable/posix/ CI groups.

samccann committed:
disable gather facts and update playbook output (#46409)

* disable facts gather and update playbook output

* fix output

s-hertel committed:
Move network utils that are used by AWS modules (#45579)

* Separate networking tools that may be used by modules outside of networking so changes to networking-only utilities don't trigger AWS integration tests

* Add unit tests for moved network utils

* Add comment to prevent imports from being mistakenly removed

* Move to_bits as well

mwellmeyer committed:
correct description

mwellmeyer committed:
correct description

ericzolf committed:
New option JUNIT_TASK_RELATIVE_PATH to output relative paths. (#37274)

* New option JUNIT_TASK_RELATIVE_PATH to output relative paths.

It makes the output in Jenkins much more readable as the absolute
path was taking the whole width of the screen, leaving no space for
the actual message.

A makedirs instead of mkdir slipped in as to make the creation of
the output directory recursive, hope this is OK.

* Add version_added field to doc set to 2.8

mattclay committed:
Fix ansible-test multi-group smoke test handling. (#46363)

* Fix ansible-test smoke tests across groups.
* Fix ansible-test list arg defaults.
* Fix ansible-test require and exclude delegation.
* Fix detection of Windows specific changes.
* Add minimal Windows testing for Python 3.7.

mwellmeyer committed:
correct description syntax

mwellmeyer committed:
correct description syntax

mwellmeyer committed:
Revert "correct description syntax"

This reverts commit 31d2621e0c8a5735d1592ea12de552dff84077a8.

mwellmeyer committed:
Revert "correct description syntax"

This reverts commit 31d2621e0c8a5735d1592ea12de552dff84077a8.

mwellmeyer committed:
Revert "correct description"

This reverts commit 6ac932f28aaab0bdf856552823f8c2f149dd7a01.

mwellmeyer committed:
Revert "correct description"

This reverts commit 6ac932f28aaab0bdf856552823f8c2f149dd7a01.

mwellmeyer committed:
revert description changes

mwellmeyer committed:
revert description changes

mwellmeyer committed:
Merge branch 'add_physdev' of https://github.com/mwellmeyer/ansible into add_physdev

mwellmeyer commented:
Moved to PR #46421
